RadixWallet/Features/CardCarousel/CardCarousel+View.swift

```diff
@@ -0,0 +1,194 @@
+import Foundation
+import SwiftUI
+
+// MARK: - CardCarousel.View
+extension CardCarousel {
+	public struct View: SwiftUI.View {
+		public let store: StoreOf<CardCarousel>
+
+		private let margin: CGFloat = .medium1
+		private let spacing: CGFloat = .small1 * 0.5
+		@ScaledMetric private var height: CGFloat = 105
+		@SwiftUI.State private var selectedCardIndex = 0
+
+		public var body: some SwiftUI.View {
+			WithPerceptionTracking {
+				VStack(spacing: 8) {
+					GeometryReader { _ in
+						coreView
+					}
+					.frame(height: store.cards.isEmpty ? 0 : height)
+
+					positionIndicator
+				}
+			}
+			.onAppear {
+				store.send(.view(.didAppear))
+			}
+		}
+
+		@MainActor
+		private var coreView: some SwiftUI.View {
+			TabView(selection: $selectedCardIndex) {
+				ForEach(Array(store.cards.enumerated()), id: \.1) { index, card in
+					CarouselCardView(card: card) {
+						store.send(.view(.cardTapped(card)))
+					} closeAction: {
+						store.send(.view(.closeTapped(card)), animation: .default)
+					}
+					.tag(index)
+					.padding(.horizontal, margin - spacing)
+					.transition(.scale(scale: 0.8).combined(with: .opacity))
+				}
+			}
+			.tabViewStyle(.page(indexDisplayMode: .never))
+			.animation(.default, value: store.cards)
+		}
+
+		private var positionIndicator: some SwiftUI.View {
+			HStack(spacing: spacing) {
+				if store.cards.count > 1 {
+					ForEach(0 ..< store.cards.count, id: \.self) { index in
+						let isSelected = selectedCardIndex == index
+						Capsule()
+							.fill(isSelected ? .app.gray4 : .app.gray2)
+							.frame(isSelected ? spacing : .small3)
+					}
+				}
+			}
+		}
+	}
+}
+
+// MARK: - CarouselCardView
+public struct CarouselCardView: View {
+	public let card: HomeCard
+	public let action: () -> Void
+	public let closeAction: () -> Void
+
+	public var body: some View {
+		ZStack(alignment: .topTrailing) {
+			Button(action: action) {
+				VStack(alignment: .leading, spacing: .small2) {
+					HStack {
+						Text(title)
+							.textStyle(.body1Header)
+							.minimumScaleFactor(0.8)
+
+						if showLinkIcon {
+							Image(.iconLinkOut)
+								.foregroundStyle(.app.gray2)
+						}
+					}
+
+					Text(text)
+						.lineSpacing(-20)
+						.lineLimit(nil)
+						.minimumScaleFactor(0.8)
+						.textStyle(.body2Regular)
+				}
+				.multilineTextAlignment(.leading)
+				.foregroundStyle(.app.gray1)
+				.padding([.top, .leading], .medium2)
+				.padding(.trailing, trailingPadding)
+				.padding(.bottom, .small1)
+				.frame(maxWidth: .infinity, maxHeight: .infinity, alignment: .topLeading)
```

Comment:
This is like a lot of padding + frame, is it really needed?

Reply:
Yes 🫤

Reply:
This makes it so that the frame grows as much as it can, instead of hugging the content. It's very similar to using say an HSTack with Spacer's.